dask/bytes/core.py

```diff
@@ -141,7 +143,7 @@ def write_bytes(data, urlpath, name_function=None, compression=None,
         raise NotImplementedError("Unknown protocol for writing %s (%s)" %
                                   (protocol, urlpath))
 
-    files = open_files_write(paths, **storage_options)
+    files = [(open_files_write, p, storage_options) for p in paths]
     out = [delayed(write_block_to_file)(v, f, compression, encoding)
            for (v, f) in zip(data, files)]
```

Comment:
Interesting that you're using what is almost a dask tuple-style task here.  We could also drop into nested dictionary mode if we wanted to keep `write_block_to_file` as it was:

```python
keys = ['write-%s' % tokenize(...) for path in paths]
return [Delayed(key, dasks=[{key: (write_block_to_file, v, 
                                    (apply, open_files_write, (p,), storage_options), 
                                    compression, encoding) ]) 
                for key, v, f in zip(keys, data, files)]
```